pkg/controller/csi_extension_server.go

```diff
@@ -0,0 +1,120 @@
+package controller
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	vgsext "github.com/dell/dell-csi-extensions/volumeGroupSnapshot"
+	"github.com/dell/gopowerstore"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// StateReady resembles ready state
+const StateReady = "Ready"
+
+// CreateVolumeGroupSnapshot creates volume group snapshot
+func (s *Service) CreateVolumeGroupSnapshot(ctx context.Context, request *vgsext.CreateVolumeGroupSnapshotRequest) (*vgsext.CreateVolumeGroupSnapshotResponse, error) {
+	log.Infof("CreateVolumeGroupSnapshot called with req: %v", request)
+
+	err := validateCreateVGSreq(request)
+	if err != nil {
+		log.Errorf("Error from CreateVolumeGroupSnapshot: %v ", err)
+		return nil, err
+	}
+	var reqParams gopowerstore.VolumeGroupSnapshotCreate
+	reqParams.Name = request.GetName()
+	reqParams.Description = request.GetDescription()
+	parsedVolHandle := strings.Split(request.SourceVolumeIDs[0], "/")
+	var arr string
+	if len(parsedVolHandle) >= 1 {
```

Comment:
it should be len() >= 2 isn't it? we are accessing the second element here.

Reply:
This is fixed in https://github.com/dell/csi-powerstore/pull/81